integration/test/ca_bundle/ca_bundle_test.go

```diff
@@ -0,0 +1,123 @@
+const agentRuntime = 1 * time.Minute
+
+
+type input struct {
+	findTarget bool
+	dataInput string
+}
+
+//Must run this test with parallel 1 since this will fail if more than one test is running at the same time
+//This test uses a pem file created for the local stack endpoint to be able to coneect via ssl
+func TestBundle(t *testing.T) {
+
+	parameters := []input{
+		//Use the system pem ca bundle  + local stack pem file ssl should connect thus target string not found
+		{dataInput: "resources/integration/ssl/with/combine/bundle", findTarget: false},
+		//Use the system pem ca bundle ssl should not connect thus target string found
+		{dataInput: "resources/integration/ssl/with/original/bundle", findTarget: true},
+		//Do not look for ca bundle should not connect thus target string found
+		{dataInput: "resources/integration/ssl/without/bundle", findTarget: true},
+	}
+
+	for _, parameter := range parameters {
+		//before test run
+		log.Printf("resource file location %s find target %t", parameter.dataInput, parameter.findTarget)
+		clearLogFile()
+		t.Run(fmt.Sprintf("resource file location %s find target %t", parameter.dataInput, parameter.findTarget), func(t *testing.T) {
+			copyFile(parameter.dataInput + configJSON, configOutputPath)
+			copyFile(parameter.dataInput + commonConfigTOML, commonConfigOutputPath)
+			startAgent();
+			time.Sleep(agentRuntime);
+			log.Printf("Agent has been running for : %s", agentRuntime.String());
+			stopAgent();
+			containsTarget := readOutputLog()
+			if (parameter.findTarget && !containsTarget) || (!parameter.findTarget && containsTarget) {
+				t.Errorf("Find target is %t contains target is %t", parameter.findTarget, containsTarget)
+			}
+		})
+	}
+}
+
+func clearLogFile() {
+	cmd := exec.Command("bash", "-c", "sudo rm " + outputLog + " && sudo touch " + outputLog)
+	err := cmd.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("Cleared log file : %s", outputLog);
+}
+
+func copyFile(pathIn string, pathOut string) {
+	log.Printf("Copy File %s to %s", pathIn, pathOut)
+	pathInAbs, err := filepath.Abs(pathIn)
+	log.Printf("File %s abs path %s", pathIn, pathInAbs)
+	if err != nil {
+		log.Fatal(err)
+	}
+	cmd := exec.Command("bash", "-c", "sudo cp " + pathInAbs + " " + pathOut)
+	err = cmd.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Printf("File : %s copied to : %s", pathIn, pathOut);
+}
+
+func startAgent() {
+	cmd := exec.Command("bash", "-c", "sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a " +
+		"fetch-config -m ec2 -s -c file:" +
+		configOutputPath)
+	err := cmd.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("Agent has started");
+}
+
+func stopAgent() {
+	cmd := exec.Command("bash", "-c", "sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a stop")
+	err := cmd.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("Agent is stopped");
+}
+
+func readOutputLog() bool {
```

Comment:
Consider renaming this function to make the purpose clearer. Wouldn't expect `readOutputLog` to return a boolean. Maybe `outputLogContainsTarget`.